src/enumerable.cr

```diff
@@ -493,6 +493,21 @@ module Enumerable(T)
     nil
   end
 
+  # Returns the last element in the collection. Raises `Enumerable::EmptyError`
+  # if the collection is empty.
+  def last
+    raise Enumerable::EmptyError.new unless any? { true }
```

Comment:
Thus traverses the collection twice. You should use uninitialized, a flag, and traverse once. There are other similar methods that do this, like max.

Reply:
Since this whole approach is not especially effective than gaining **one** iteration doesn't seem worth to me going all fancy with optimizations here...

Reply:
It's not an optimizaton. Enumerable methods implicitly guarantee that the collection is traversed just once.

If you want, I can send a separate PR to implement it.

Reply:
Done, have a look is it what you had in mind.

Reply:
It looks good!

> Didn't know about that. Where's it stated btw?

It's not stated, but for stdlib tries it's best to avoid repeating logic.

I think there's no need for `last_internal`. For `last` we need to use `uninitialized`. But for `last?` we can do it like this:

```crystal
def last?
  obj = nil
  each { |e| obj = e }
  obj
end
```

Reply:
Done and done.